lib/binaries.sh

```diff
@@ -48,7 +48,11 @@ install_yarn() {
 
   read -r number url < <(echo "$resolve_result")
 
-  echo "Downloading and installing yarn ($number)..."
+  if [[ -n "$YARN_BINARY_URL" ]]; then
+    url="$YARN_BINARY_URL"
+  fi
+
+  echo "Downloading and installing yarn ($number) from $url"
```

Comment:
I think we want to leave alone the behavior for people that install via the default binary url, and only let the user that has specified a url.

So the behavior will look like this..

if there's a url in the ENV var:
  install from the url
  log that we are downloading and installing from the url
else:
  resolve from the version (Line 43)
  read the resolved number and url (Line 49)
  log that we are downloading and installing from yarn version

Reply:
I can change it to follow this pseudocode but does it not do the same thing? I have the url being set to the default binary url and then overwritten if the env var is set.

Reply:
also let's say that the user specifies an invalid url, should we check for this and install using the default url instead?

Reply:
It does, but there's 2 things that I think should be different:
1. We don't need to call `resolve` because the reason that we make that call (to S3) is to be able to hit the right url. This is an extra network call that doesn't need to be made, so we should stop it from happening.
2. We don't need to log the url and version number for both: we will want to log the url for the custom urls, and just the version number for the users that don't use the urls. This is because the users with custom urls are specifying their urls and the users that are not are specifying versions, so we just need to give the minimum amount of information to the user (either url or version) while also being explicit in what we are downloading.

Reply:
If the user specifies an invalid url, the build should fail

Reply:
(we can add a failure here, but the buildpack will error anyways if it's an invalid url that will probably provide a download failure that the user will probably be able to deduce what is wrong with. If we find that users aren't able to figure it out, we can catch the error and fail.)